velox/functions/sparksql/String.h

```diff
@@ -1181,4 +1181,63 @@ struct FindInSetFunction {
   }
 };
 
+template <typename T>
+struct SoundexFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  static constexpr bool is_default_ascii_behavior = true;
+  static constexpr int32_t reuse_strings_from_arg = 0;
+
+  void call(out_type<Varchar>& result, const arg_type<Varchar>& input) {
```

Comment:
Can we put the reference of implementation to the PR description?
https://github.com/apache/spark/blob/master/common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java#L1793-L1838